### DISCUSSION TITLE
Who have a configuration to run in turborepo monorepo ?

### DISCUSSION THREAD

**Discussion by Suniron** [orta/vscode-jest#1015] [Q&A]
I'm looking for a configuration to put in my `.vscode/settings.json`.

I have a monorepo powered by `Turborepo`, so my tests are in `apps/*/__tests__` and `packages/*/__tests__` and i can't find the config to use the **vscode-jest extension** in my project.

All tests works in CLI ;).

Thank's :)

**Comment by felipedeboni** [comment]
@Suniron I think I've finally made it work! Please note it is far from being pretty and you will have to add all your individual packages to the `jest.virtualFolders` key, but at least it's working.

In your root directory, add or edit `.vscode/settings.json` file and put the following content:

```json
{
  "jest.autoRun": "off",
  "jest.jestCommandLine": "pnpm test --",
  "jest.rootPath": ".",
  "jest.debugMode": true,
  "jest.virtualFolders": [
    {
      "name": "packages/your-package-name",
      "jestCommandLine": "cd packages/your-package-name && pnpm test --"
    }
  ]
}
```

Notes:

- Package manager is `PNPM@8.7.4` 
- `ts-node` reads the configs and `@swc/jest` transpiles code
- `Jest` and it's dependencies are installed on workspace root
- Each package is going to have it's own `jest.config.ts`
- It's not multi-root and I don't have `*.code-workspace` file.
- I tried debugging a test, however it's broken and since it's really late here I am going to solve it another day.
- Probably it's a good idea to create a script to update `jest.virtualFolders` when needed.

Good luck 😄
